packages/components/src/input-control/reducer/reducer.ts

```diff
@@ -54,6 +54,15 @@ function inputControlStateReducer(
 	return ( state, action ) => {
 		const nextState = { ...state };
 
+		// Updates state and returns early when there's no action type. These
+		// are controlled updates and need no exposure to additional reducers.
+		if ( ! ( 'type' in action ) ) {
+			let { value = state.value } = action;
+			value ??= '';
+			if ( value !== '' ) value = `${ value }`;
+			return { ...state, value };
```

Comment:
Nit: As a code reader I would find a simpler version like this to be easier to verify the logic 👇

```suggestion
			return {
				...state,
				value: `${ action.state.value ?? '' }`,
			};
```

(Or something like that, anything that removes the let/if)